### PR TITLE
[#114] Sign and hash artifacts

### LINKED ISSUE #114: Sign artifacts and provide hash sums
# Clarification and motivation

People want to be more confident that artifacts they use can be trusted. E. g. if I obtain our static `tezos-client` somehow, I want to be sure it was provided by Serokell and is genuine as long as I trust Serokell (or, if we move packaging to `tezos/tezos`, trust Tezos developers).

There are at least two natural ways to do that:
1. Sign each artifact using our key (e. g. we can generate a fresh key specifically for this repo), provide our public key somewhere, provide signatures along with artifacts.
2. Provide hash sum (e. g. sha256) for each artifact.

As an example, I see that `stack` releases include `.asc` and `.sha256` files for each artifact: https://github.com/commercialhaskell/stack/releases/tag/v2.5.1

For `.deb` and `.rpm` packages it might be (and actually is quite likely) possible to include signatures into packages themselves. We should explore this opportunity.

# Acceptance criteria

1. We provide at least one way to (quickly and easily) verify our artifacts.
2. For packages we use some standard approach if it exists.

### PR DESCRIPTION
**Pull request by rvem**
## Description
We want to provide some way for people to check whether binaries/packages they're using are genuine
and created by Serokell.

This PR adds sha256 sum calculation along with PGP signing for all autorelease artifacts.

Also, this PR adds instruction about how native packages should be signed before publishing.

## Related issue(s)

Resolves #114

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -12,6 +12,8 @@ let
     Binaries that target arm64 architecture has `-arm64` suffix in the name.
     Other binaries target x86_64.
 
+    Release artifacts are signed with the following key: 0x7EAF9B150ACE940CF8C008A0BF847A85AC7BF43E
```
Comment (gromakovsky) [inline release.nix:15]: Perhaps we could also suggest a keyserver from which this key can be fetched.

File: scripts/autorelease.sh [hunk 2]
```diff
@@ -22,12 +22,13 @@ assets_dir=$TEMPDIR/assets
...
           --arg docker-binaries ./binaries/docker --arg docker-arm-binaries ./arm-binaries/docker
 mkdir -p "$assets_dir"
-for asset in "$assets_dir"/*; do
-    sha256sum "$asset" > "$asset.sha256"
-done
 # Move archive with binaries and tezos license to assets
 shopt -s extglob
 cp -L "$TEMPDIR"/"$project"/!(*.md) "$assets_dir"
+for asset in $assets_dir/*; do
+    sha256sum "$asset" | sed 's/ .*/ /' > "$asset.sha256"
```
Comment (gromakovsky) [inline scripts/autorelease.sh:29]: What is this `sed` for?

File: scripts/autorelease.sh [hunk 3]
```diff
@@ -22,12 +22,13 @@ assets_dir=$TEMPDIR/assets
 nix-build -A release -o "$TEMPDIR"/"$project" --arg timestamp "$(date +\"%Y%m%d%H%M\")" \
           --arg docker-binaries ./binaries/docker --arg docker-arm-binaries ./arm-binaries/docker
 mkdir -p "$assets_dir"
-for asset in "$assets_dir"/*; do
-    sha256sum "$asset" > "$asset.sha256"
-done
 # Move archive with binaries and tezos license to assets
 shopt -s extglob
 cp -L "$TEMPDIR"/"$project"/!(*.md) "$assets_dir"
+for asset in $assets_dir/*; do
```
Comment (gromakovsky) [inline scripts/autorelease.sh:28]: Looks like most of the steps here are commented, so I'd leave a comment above as well.

File: scripts/autorelease.sh (hunk 2 above)
Comment (rvem) [reply scripts/autorelease.sh:29]: strip `/tmp/bla-bla` suffix from the hash message

**Review by gromakovsky — APPROVED** [review]

**Pull request merged**